src/m_gsm_symbolic/sofie_test_parser.py

```diff
@@ -0,0 +1,225 @@
+
+        # hmmm might not be the smartest solution? But I couldnt replace variable values for the 
+        # answer annoted, if I didnt extract the default dict from it? 
+        annotated_question_answer = self.question_annotated + self.answer_annotated
+
+        matches = re.findall(regex, annotated_question_answer) # returns a list containing all matches
+        for var_name, value in matches:
+            default_values[var_name] = value
+        return default_values
+
+    @staticmethod
+    def remove_calculations(text: str) -> str:
+        """remove calculations from the text. re.sub() "replaces" the calculations with noting"""
+        return re.sub(r"<<.*?>>", "", text)  
+    
+    def apply_init_rules(self,
+                        names: list[str],
+                        multiple_ice: list[int],
+                        multi_times: list[int],
+                        functions: dict[str, callable]) -> dict[str, str]:
+
+        """apply init rules to generate new variable values"""
+        new_values = {}
+
+        for line in self.init.strip().splitlines(): # split str into seperate lines at line break "\n"
+            line = line.lstrip("-").strip() # remove the leading "-" character + space
+            var, rule = line.split("=") # split var (left side of =) and "rule" (right side of =) by "="
+            var = var.strip().lstrip("$") # remove space at the beginning + chr $ in front of the variables
+            rule = rule.strip() # again, remove space at the beginning of the string
+
+            if "range" in rule:
+                range_arguments = list(map(int, re.findall(r"\d+", rule))) # find range of values, digit chr
+                if len(range_arguments) == 2: #hmm might not be the most "stable" solution
+                    start, stop = range_arguments
+                    new_values[var] = str(random.choice(functions["range"](start, stop)))
+                elif len(range_arguments) == 3:
+                    start, stop, step = range_arguments
+                    new_values[var] = str(random.choice(functions["range"](start, stop, step)))
+                
+            elif "sample" in rule:
+                match_list = re.search(r"sample\((.+?)\)", rule)
+                sample_expression = match_list.group(1)
+
+                sampled_arguments = eval(f"functions['sample']({sample_expression})", {"names": names,
+                                                                                       "multiple_ice": multiple_ice,
+                                                                                       "multi_times": multi_times,
+                                                                                       "functions": functions})
+                var_arguments = [v.strip() for v in var.split(",")]
+
+                for var_sample, val_sample in zip(var_arguments, sampled_arguments):
+                    print(f"Assigning {var_sample} = {val_sample}") 
```

Comment:
I would generally use the [`logging`](https://docs.python.org/3/howto/logging.html) library for test prints. (it is a part of python)